src/nvim/lua/treesitter.c

```diff
@@ -646,6 +654,36 @@ static int node_symbol(lua_State *L)
   return 1;
 }
 
+static int node_field(lua_State *L)
+{
+  TSNode node;
+  if (!node_check(L, 1, &node)) {
+    return 0;
+  }
+
+  size_t name_len;
+  const char *field_name = luaL_checklstring(L, 2, &name_len);
+
+  TSTreeCursor cursor = ts_tree_cursor_new(node);
+
+  lua_newtable(L);  // [table]
+  unsigned int curr_index = 0;
+
+  if (ts_tree_cursor_goto_first_child(&cursor)) {
+    // TODO(vigoux): It seems that multiple fields with the same name
+    //  are possible, but is that really a good practice ?
```

Comment:
Do we need this TODO here?

If you're sure you only need one, we could optionally pass in another variable (`find_first`) and then it doesn't return a list, it just returns the node. And it quits out earlier.

Reply:
Maybe we don't need it but I find it weird to hav multiple node with the same field

Reply:
You can just return multiple values, then only callers looking for more values would see them or capturing them in a table would see them.